src/metabase/models/setting.clj

```diff
@@ -313,6 +314,18 @@
   set to true when settings are being written directly via /api/setting endpoints."
   false)
 
+(defn- has-advanced-setting-access?
+  "If `advanced-permissions` is enabled, check if current user has permissions to edit `setting`.
+  Return `false` when `advanced-permissions` is disabled."
+  []
+  (u/ignore-exceptions
+   (classloader/require 'metabase-enterprise.advanced-permissions.common
+                        'metabase.public-settings.premium-features))
+  (if (and ((resolve 'metabase.public-settings.premium-features/enable-advanced-permissions?))
+           (resolve 'metabase-enterprise.advanced-permissions.common/current-user-has-general-permissions?))
+    ((resolve 'metabase-enterprise.advanced-permissions.common/current-user-has-general-permissions?) :setting)
```

Comment:
nit: `resolve` will return the resolved function or `nil`...probably be cleaner to use something like `if-let` here and try to avoid calling `resolve` twice